internal/marshaled/source.go

```diff
@@ -198,53 +211,92 @@ func (mds *DataSource) WatchItems(formatName string, iw source.ItemWatcher) erro
 
 func (mds *DataSource) startWatching() error {
```

Comment:
Worth mentioning in a comment here that a lock isn't needed here because this is only called from contexts where the lock is already acquired.